### PR TITLE
Import new Bionic module from Android overlay instead where possible

### PR DESCRIPTION
**Pull request by finagolfin**
Also, add the import in `Environment.swift` too

@rauhul, you added that last file just after I got these imports in with #7615, so I recently had to add this too to keep it building on Android.

### CODE REVIEW

File: Sources/Basics/Environment/Environment.swift [hunk 1]
```diff
@@ -19,6 +19,8 @@ import Musl
 #elseif os(Windows)
 import CRT
 import WinSDK
+#elseif canImport(Android)
+import Android
```
Comment (compnerd) [inline Sources/Basics/Environment/Environment.swift:23]: Can we use `Bionic` here instead? Note that Darwin is using `Darwin.C` and Windows is using `CRT`. Of course if we need more than the C interfaces, `Android` is fine, but would bee nice to keep this scoped to the same level as the other platforms.

Comment (finagolfin) [reply Sources/Basics/Environment/Environment.swift:23]: I'll check if that's enough.

Comment (finagolfin) [reply Sources/Basics/Environment/Environment.swift]: Done

**Comment by finagolfin** [comment]
This now makes the changes suggested in #7752, along with the original change here modified to `Bionic`. All except one test fixture from #7615 worked with `import Bionic`.

@compnerd, please review, since you asked for this change.

**Review by compnerd — APPROVED** [review]

**Comment by finagolfin** [comment]
@rauhul, a CI run and we can get this in.

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by finagolfin** [comment]
@kateinoigakukun, passed CI, ready for merge.

**Pull request merged**